lib/taskana-core/src/main/java/pro/taskana/impl/TaskServiceImpl.java

```diff
@@ -759,29 +719,147 @@ private void standardSettings(TaskImpl task, Classification classification,
         }
     }
 
-    List<TaskSummary> augmentTaskSummariesByContainedSummaries(List<TaskSummaryImpl> taskSummaries) {
-        LOGGER.debug("entry to augmentTaskSummariesByContainedSummaries()");
-        List<TaskSummary> result = new ArrayList<>();
-        if (taskSummaries == null || taskSummaries.isEmpty()) {
-            return result;
+    private BulkOperationResults<String, TaskanaException> transferTasks(List<String> taskIdsToBeTransferred,
+        Workbasket destinationWorkbasket)
+        throws InvalidArgumentException, WorkbasketNotFoundException, NotAuthorizedException {
+
+        workbasketService.checkAuthorization(destinationWorkbasket.getId(), WorkbasketPermission.APPEND);
+
+        if (taskIdsToBeTransferred == null) {
+            throw new InvalidArgumentException("TaskIds must not be null.");
         }
+        BulkOperationResults<String, TaskanaException> bulkLog = new BulkOperationResults<>();
+        List<String> taskIds = new ArrayList<>(taskIdsToBeTransferred);
+        removeNonExistingTasks(taskIds, bulkLog);
```

Comment:
maybe even more readable?
removeNonExistingTasksFromTaskIdList(taskIds, bulkLog);

or even better?
removeNonExistingTasksFromTaskIdListAndStoreErrorsToBuldLog(taskIds, bulkLog);

I'm still trying to figure out myself, what I do like most. What do you think?

Reply:
I prefer removeNonExistingTasksFromTaskIdList, where are storing result in the bulk log in many places and we are not refering to that action.